board.h

```diff
@@ -9,29 +9,35 @@
 #include "sphere.h"
 
 /**
- * @brief 
+ * Represents a game board containing spheres.
+ * Has functionality to allow access to spheres, as well as the
+ * ability to move gems between spheres.
  * 
+ * A board contains a sphere for each element, represented as a 
+ * map of elements to spheres.
+ * 
+ * When constructed, each sphere contains 1 gem of the same element as 
+ * the sphere. Any changes to the gems must be effected after construction.
  */
 class Board {
-  public:
-  Board();
-  // count accepts int inputs rather than Element so that it can
-  // deal with bitwise operations on inputs.
-  int count(const int gemType=Element::ALL, const int sphereType=Element::ALL) const;
-  
-  bool isMoveValid(const std::vector<std::tuple<Element, Element, Element>>) const;
-  void move(const std::vector<std::tuple<Element, Element, Element>>);
+  public: 
+    Board();
 
-  // void removeGem(const Element, const Element);
+    // count accepts int inputs rather than Element so that it can
+    // deal with bitwise operations on inputs.
+    int count(const int gemType=Element::ALL, const int sphereType=Element::ALL) const;
+  
+    // Each move in the vector is in the order {gemType, source, destination}
+    bool isMoveValid(const std::vector<std::tuple<Element, Element, Element>> moves) const;
+    void move(const std::vector<std::tuple<Element, Element, Element>> moves);
 
-  void setSphere(const Element, const Sphere);
-  Sphere getSphere(const Element) const;
+    void setSphere(const Element, const Sphere);
+    Sphere getSphere(const Element) const;
 
 
   private:
-  std::map<Element, Sphere> _spheres;
-
-  // bool isMoveValidSingle(const std::tuple<Element, Element, Element>) const;
+    // The spheres included in the board.
```

Comment:
I don't see the value in this comment. I know they are spheres from the name of the type and I know they are in the board because they are a member. I suggest removing this comment.